Nodejs/Product/Nodejs/Project/NodeModulesNode.cs

```diff
@@ -27,9 +27,35 @@
 using Microsoft.VisualStudioTools.Project;
 using MessageBox = System.Windows.MessageBox;
 using Timer = System.Threading.Timer;
+using System.Threading.Tasks;
 
 namespace Microsoft.NodejsTools.Project {
     internal class NodeModulesNode : AbstractNpmNode {
+        /// <summary>
+        /// `INpmGlobalPackageProvider` that caches `IGlobalPackages` objects for reuse.
+        /// </summary>
+        private class CacheGlobalPackageProivder : INpmGlobalPackageProvider {
+            private Dictionary<string, WeakReference<IGlobalPackages>> _packages = new Dictionary<string, WeakReference<IGlobalPackages>>();
+
+            public async Task<IGlobalPackages> GetGlobalPackages(string pathToNpm) {
+                WeakReference<IGlobalPackages> reference;
+                if (_packages.TryGetValue(pathToNpm, out reference)) {
+                    IGlobalPackages current;
+                    if (reference.TryGetTarget(out current) && current != null) {
+                        return current;
+                    }
+                }
+                var provider = new Npm.SPI.NpmBinGlobalPackageProvider();
```

Comment:
the additional level of abstraction of a bin global package provider doesn't seem necessary here. there's nothing special about "bin" - that's just how we're getting the location of the global packages.